### PR TITLE
Unpack the base queue

### PR DESCRIPTION
**Pull request by treeowl**
* Unpack the base queue into the catenable queue.
* Stop using type-aligned stuff; we don't really need it.

### CODE REVIEW

**Comment by treeowl** [comment]
These seems to improve almost everything moderately to slightly. The exception is that the `interleave/L.SeqT_S` tests get a little bit slower for some reason.

**Comment by treeowl** [comment]
The blasted Haddock internal error is back. @dagit, I propose either dropping the Windows runner so I can exclude Haddock from the CI or dropping support for GHC before 7.8. Personally, I'd love to stop having to futz with antiquated GHC compat, but you seem pretty attached to it.

**Comment by treeowl** [comment]
@dagit can you reproduce my benchmark results?

**Comment by treeowl** [comment]
OK, the Haddock nonsense is gone, for no particularly obvious reason. Windows is still messed up, but that's not my fault.

**Comment by dagit** [comment]
> I propose either dropping the Windows runner

Did my earlier fix to make it optional not work? Sorry. I'll look into it more.

I might be slow to respond this week and next. Trying to take a vacation before starting a new job.

File: src/Control/Monad/Logic/Sequence/Internal/ScheduledQueue.hs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+infixl 5 :>
+-- | A strict-spined snoc-list
+data SL a
+  = SNil
+  | !(SL a) :> a
+  deriving (Functor, Foldable, Traversable)
+
+revAppend :: [a] -> SL a -> [a]
+revAppend l r = rotate l r []
```
Comment (treeowl) [inline src/Control/Monad/Logic/Sequence/Internal/ScheduledQueue.hs:49]: Rename this function.

**Comment by treeowl** [comment]
BTW, I've offered PRs with the fix and improvements to `FastQueue` upstream https://github.com/atzeus/sequence/pull/3, but the package doesn't really seem to be active, so I don't know if that'll go anywhere.

**Comment by treeowl** [comment]
@dagit, I just found a strictness bug in the upstream version which I'm fixing as we speak. Might be interesting to re-run your benchmarks on this, though I can't say whether you were using persistence anyway.

**Comment by treeowl** [comment]
Turns out the strictness bug was already fixed in `type-aligned`, but not in `sequence`. I used the fix from `type-aligned`.

**Review by dagit — APPROVED** [review]
This looks good to me. I haven't tried running the benchmark, but I'd say merge it. I doubt it's going to make things worse.

**Pull request merged**